### PR TITLE
chore: move to WinUI/WinAppSDK

### LINKED ISSUE #399: [Gallery] Migrate to WinUI/WinAppSDK
### Current behavior

progressring ‎- Photos
Lottie ‎- Photos

### Expected behavior

progressring correct
Lottie ‎correct

### How to reproduce it (as minimally and precisely as possible)

1. Open Uno Gallery APP
2. Click Burger Menu on the left top right.
3. Click "Progress Ring/Bar" from the "Component" list.
4. Observe that  'ProgressRing/Bar'  is missing.
5. Click "Lottie" from the "Features" list.
6. Observe that  'Lottie'  is missing.

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

iOS, Android

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by kazo0**
closes #399

### CODE REVIEW

File: Uno.Gallery/Uno.Gallery.Shared/ViewModels/Command.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Uno.Gallery.ViewModels
 #if IS_WINUI
 		[Microsoft.UI.Xaml.Data.Bindable]
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/ViewModels/Command.cs:10]:
```suggestion
	[Microsoft.UI.Xaml.Data.Bindable]
```

File: Uno.Gallery/Uno.Gallery.Shared/Views/NestedPages/MediaPlayerElementSample_NestedPage4.xaml [hunk 2]
```diff
@@ -59,8 +60,8 @@
 														IsFastForwardEnabled="{Binding IsChecked, ElementName=IsFastForwardEnabled}"
 														IsFastRewindButtonVisible="{Binding IsChecked, ElementName=IsFastRewindButtonVisible}"
 														IsFastRewindEnabled="{Binding IsChecked, ElementName=IsFastRewindEnabled}"
-														IsFullWindowButtonVisible="{Binding IsChecked, ElementName=IsFullWindowButtonVisible}"
-														IsFullWindowEnabled="{Binding IsChecked, ElementName=IsFullWindowEnabled}"
+														not_win:IsFullWindowButtonVisible="{Binding IsChecked, ElementName=IsFullWindowButtonVisible}"
+														not_win:IsFullWindowEnabled="{Binding IsChecked, ElementName=IsFullWindowEnabled}"
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/NestedPages/MediaPlayerElementSample_NestedPage4.xaml:64]: if you are going to comment out no-existing properties for windows
make sure you also do the same for the option buttons controlling them

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/AutoSuggestBoxSamplePage.xaml.cs [hunk 3]
```diff
@@ -2,11 +2,11 @@
 using System.Collections.Generic;
 using System.Linq;
 using Uno.Gallery.ViewModels;
-using Windows.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls;
 
 namespace Uno.Gallery.Views.Samples
 {
-	[SamplePage(SampleCategory.UIComponents, "AutoSuggestBox", Description = "Represents a text control that makes suggestions to users as they enter text.", DocumentationLink = "https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.autosuggestbox", DataType = typeof(AutoSuggestBoxSamplePageViewModel))]
+	[SamplePage(SampleCategory.UIComponents, "AutoSuggestBox", Description = "Represents a text control that makes suggestions to users as they enter text.", DocumentationLink = "https://learn.microsoft.com/en-us/uwp/api/Microsoft.UI.Xaml.controls.autosuggestbox", DataType = typeof(AutoSuggestBoxSamplePageViewModel))]
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/AutoSuggestBoxSamplePage.xaml.cs:9]: 404 deadlink
```suggestion
	[SamplePage(SampleCategory.UIComponents, "AutoSuggestBox", Description = "Represents a text control that makes suggestions to users as they enter text.", DocumentationLink = "https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.autosuggestbox", DataType = typeof(AutoSuggestBoxSamplePageViewModel))]
```

Comment (Xiaoy312) [reply Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/AutoSuggestBoxSamplePage.xaml.cs:9]: please review similar links that you globally replaced...

File: Uno.Gallery/Uno.Gallery.Shared/Controls/AppDialog.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Uno.Gallery
+{
+    public partial class AppDialog : ContentDialog
+	{
+		public AppDialog()
+		{
+			XamlRoot = App.Instance.MainWindow.Content.XamlRoot;
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Controls/AppDialog.cs:12]: some justification/explanation could be helpful here

File: Uno.Gallery/Uno.Gallery.Windows/MainWindow.xaml
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Windows/MainWindow.xaml:1]: remove along with cs; no use

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/SharingSamplePage.xaml.cs [hunk 5]
```diff
@@ -129,4 +133,47 @@ private void DataRequested_TextLight(DataTransferManager sender, DataRequestedEv
 			sender.DataRequested -= DataRequested_TextLight;
 		}
 	}
+
+	public static class DataTransferManagerHelper
+	{
+		static readonly Guid _dtm_iid = new Guid(0xa5caee9b, 0x8708, 0x49d1, 0x8d, 0x36, 0x67, 0xd2, 0x5a, 0x8d, 0xa0, 0x0c);
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/SharingSamplePage.xaml.cs:139]:
```suggestion
		private static readonly Guid _dtm_iid = new Guid("a5caee9b-8708-49d1-8d36-67d25a8da00c");
```

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/PowerManagerSamplePage.xaml.cs [hunk 6]
```diff
@@ -109,9 +109,9 @@ public bool MonitoringPowerStatus
 			}
 		}
 
-		protected async void OnPropertyChanged([CallerMemberName] string propName = "")
+		protected void OnPropertyChanged([CallerMemberName] string propName = "")
 		{
-			await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+			App.Instance.MainWindow.DispatcherQueue.TryEnqueue(Microsoft.UI.Dispatching.DispatcherQueuePriority.Normal, () =>
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/PowerManagerSamplePage.xaml.cs:114]: please add an using

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/MediaPlayerElementSamplePage.xaml.cs [hunk 7]
```diff
@@ -1,37 +1,37 @@
...
+	[SamplePage(SampleCategory.UIComponents, "MediaPlayerElement", Description = "Represents an object that uses a MediaPlayer to render audio and video to the display.", DocumentationLink = "https://learn.microsoft.com/en-us/uwp/api/Microsoft.UI.Xaml.controls.mediaplayerelement")]
 	public sealed partial class MediaPlayerElementSamplePage : Page
 	{
 		public MediaPlayerElementSamplePage()
 		{
 			this.InitializeComponent();
 		}
 
-		private void LaunchMediaPlayerElementSample1(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+		private void LaunchMediaPlayerElementSample1(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/MediaPlayerElementSamplePage.xaml.cs:14]: please add an using

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/GamepadSamplePage.xaml.cs [hunk 8]
```diff
@@ -98,7 +98,7 @@ private void GamepadsChanged(object sender, Gamepad e)
 
 		private async void UpdateGamepads()
 		{
-			await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
+			App.Instance.MainWindow.DispatcherQueue.TryEnqueue(Microsoft.UI.Dispatching.DispatcherQueuePriority.Normal, () =>
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/GamepadSamplePage.xaml.cs:101]:
```suggestion
			_ = App.Instance.MainWindow.DispatcherQueue.TryEnqueue(DispatcherQueuePriority.Normal, () =>
```
\+ using

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/FileFolderPickerSamplePage.xaml.cs [hunk 9]
```diff
@@ -22,28 +23,36 @@ private async void PickFileButton_Click(object sender, RoutedEventArgs e)
 		{
 			var picker = new FileOpenPicker();
 			picker.FileTypeFilter.Add("*");
+#if WINDOWS
+			var hwnd = WindowNative.GetWindowHandle(App.Instance.MainWindow);  // App.m_window?
+			 InitializeWithWindow.Initialize(picker, hwnd);
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/FileFolderPickerSamplePage.xaml.cs:28]: add something like
```cs
private void InitForWin(object instance) // `object` here can be replaced by whatever type of 1st param of InitializeWithWindow.Initialize
{
#if WINDOWS
	var handle = WindowNative.GetWindowHandle(App.Instance.MainWindow);
	InitializeWithWindow.Initialize(instance, handle);
#endif
}
```

and call that instead

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/EmailManagerSamplePage.xaml.cs [hunk 10]
```diff
@@ -51,13 +51,16 @@ private async void Compose()
...
-				ContentDialog errorDialog = new ContentDialog
-				{
-					Title = "Error",
-					Content = "Error trying to fire show compose new email: " + Environment.NewLine + ex.Message,
-					CloseButtonText = "OK"
-				};
-				await errorDialog.ShowAsync();
+
+				
+					ContentDialog errorDialog = new ContentDialog
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/EmailManagerSamplePage.xaml.cs:56]: not using the custom AppDialog?

File: Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/GridViewSamplePage.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+			this.InitializeComponent();
+		}
+	}
+	public class GridViewSampleViewModel : ViewModelBase
+	{
+		public char[] Items { get => GetProperty<char[]>(); set => SetProperty(value); }
+
+		public GridViewSampleViewModel()
+		{
+			this.Items = "123456781234567812345678".ToCharArray();
```
Comment (Xiaoy312) [inline Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/GridViewSamplePage.xaml.cs:33]: dont use duplicated items
make it alphanumeric if you need that many items

**Review by agneszitte — COMMENTED** [review]
@kazo0 I grabbed the PR branch and can't see Windows.Package to be able to launch the Windows WinUI version
image

**Review by agneszitte — APPROVED** [review]

**Comment by kazo0** [comment]
Current issues:

- x64 build of Windows fails because of https://github.com/CommunityToolkit/Lottie-Windows/issues/515
- App fails to build in release for net7.0-ios as a result of https://github.com/dotnet/runtime/issues/90032

**Comment by kazo0** [comment]
https://github.com/Mergifyio backport legacy/4x

**Pull request merged**